Omstrukturere innhold (#30)
navikt/nada: docs/prosessere-data/notebooks.md

## Maskintype og GPU
Det er du som vet best hva du trenger, derfor er det ingen begrensninger på hva du kan velge av maskin og GPU.
Bare husk at det kan bli veldig kostbart hvis du lar en maskin (med mye minne og GPU) stå uten at den blir brukt.

## Stoppe en maskin
Har du en maskin du ikke trenger for en periode, men ikke ønsker å [slette](#slette-en-maskin), så kan du stoppe den.
Velg maskinen i [oversikten](https://console.cloud.google.com/ai-platform/notebooks/instances) og trykk `STOP` i toppen.

## Slette en maskin
Har du en maskin du ikke trenger lenger, så kan du slette den.
Velg maskinen i [oversikten](https://console.cloud.google.com/ai-platform/notebooks/instances) og trykk `DELETE` i toppen.

## Autentisering med brukers credentials på serveren

1. I Jupyterlab, åpne en terminal
2. Kjør kommandoen `gcloud auth application-default login`
3. Gå til lenken som vises i terminalen og logg inn med NAV-bruker
4. Etter at du har logget inn kopierer du verifikasjonskoden du får inn i terminalen

> Etter å ha utført stegene over vil du i din Notebook kunne jobbe med dine private Google credentials mot kilder dersom du har behov for dette.

Reviewer: Hvorfor legge til dette med behov? Tenker du at folk gjør disse stegene uansett? Burde vi heller hatt en introduksjon til dette segmentet som forteller hvorfor man skulle ha gjort de følgende stegene?

Author: Har forstått det som at en ikke nødvendigvis trenger å autentisere seg dersom en har notebookserver med servicebruker, servicebrukeren har nok rettigheter og en har riktige hemmeligheter. Jeg misforsto når jeg leste dette avsnittet og trodde jeg måtte autentisere meg, men egentlig så sto det godt nok fordi det står hva en oppnår med å gjøre det. Så den endringen jeg foreslo er nok egentlig ikke så nødvendig likevel.

> Etter å ha utført stegene over vil du i din Notebook kunne jobbe med dine private Google credentials mot kilder dersom du har behov for dette.

Reviewer: ```suggestion
Etter å ha utført stegene over vil du i din Notebook kunne jobbe med dine private Google credentials mot andre tjenester i GCP.
```

Denne tilgangen er kun midlertidig, og man må gjøre dette hver dag.

## Credentials og hemmeligheter.
Det anbefales å bruke [Secret manager](https://console.cloud.google.com/security/secret-manager?project=knada-gcp) som erstatning for Vault.

1. Gå til [Secret manager](https://console.cloud.google.com/security/secret-manager?project=knada-gcp)

> 2. Legg inn relevante verdier i din hemmelighet. Velg `NEW VERSION` og last opp en fil med hemmeligheter eller skriv inn hemmeligheter.

Reviewer: ```suggestion
2. Velg `NEW VERSION` og last opp en fil med hemmeligheter eller skriv inn hemmelighetene dine.
```

For å bruke IP i stedet for en vanlig adresse så er det bare å bytte direkte.
Så hvis følgende er den "vanlige" kode-snutten din:
```python
dsnStr = cx_Oracle.makedsn('dm08db01.adeo.no','1521',service_name='DWH_HA')
```

så kan du heller skrive
```python
dsnStr = cx_Oracle.makedsn('10.x.x.x','1521',service_name='DWH_HA')
```

## Maskintype og GPU
Det er du som vet best hva du trenger, derfor er det ingen begrensninger på hva du kan velge av maskin og GPU.
Bare husk at det kan bli veldig kostbart hvis du lar en maskin (med mye minne og GPU) stå uten at den blir brukt.

## Stoppe en maskin
Har du en maskin du ikke trenger for en periode, men ikke ønsker å [slette](#slette-en-maskin), så kan du stoppe den.
Velg maskinen i [oversikten](https://console.cloud.google.com/ai-platform/notebooks/instances) og trykk `STOP` i toppen.

## Slette en maskin
Har du en maskin du ikke trenger lenger, så kan du slette den.
Velg maskinen i [oversikten](https://console.cloud.google.com/ai-platform/notebooks/instances) og trykk `DELETE` i toppen.

## Oppsett for Oracle og Postgres drivere

> For å gjøre det enkelt for dere å komme i gang, har vi lagd to scripts som begge må kjøres med root privilegier.

Author: Legge til en setning om hva drivere er/ hvorfor dette er nyttig å gjøre?

Reviewer: Merker jeg er veldig usikker på hvilket nivå jeg skal legge meg på når det gjelder spørsmålet "hva drivere er". Kunne det holdt å lenke til https://en.wikipedia.org/wiki/Device_driver?
> hvorfor dette er nyttig å gjøre?
Lurer du på hvorfor det er nyttig å ha en driver?

Kan dette være nok?
> (en driver (i vår kontekst) er et program man må ha for å snakke med en database)

Author: Synes forslaget ditt er nok for å forklare hva det er 👍 . Tenker det kan være greit for de som ikke er kjent med begrepet at det forklares siden det står i tittelen, og at det kan forklare eller gjøre "hvorfor" tydeligere. Når jeg leste dokumentasjonen forsto jeg ikke hvorfor jeg burde gjøre dette/ hva skriptene gjør.

> For å gjøre det enkelt for dere å komme i gang, har vi lagd to scripts som begge må kjøres med root privilegier. Scriptene bør kjøres etter at driver for Oracle eller postgres er installert.

Reviewer: ```suggestion
For å gjøre det enkelt for dere å komme i gang, har vi lagd to scripts som begge må kjøres med root privilegier.
Scriptene må kjøres etter at driver for Oracle eller postgres er installert.
En driver (i vår kontekst) er et program man må ha for å snakke med en database.
```
